specs/src/specs/namespace.md

```diff
@@ -34,14 +48,27 @@ A namespace with version `0` must contain an id with a prefix of 18 leading `0`
 0x1111111111111111111111111111111111111111111111111111111111 // invalid because it does not have version 0
 ```
 
-A new namespace version MUST be introduced if the namespace format changes in a backwards incompatible way (i.e. the number of leading `0` bytes in the id prefix is reduced).
+Any change in the number of leading `0` bytes in the id of a namespace with version `0` is considered a backwards incompatible change and MUST be introduced as a new namespace version.
 
 ### ID
 
-The namespace ID is a 28 byte identifier that uniquely identifies a namespace. The ID is encoded as a byte slice of length 28.
+The namespace ID is a 28 byte identifier that uniquely identifies a namespace.
+The ID is encoded as a byte slice of length 28.
+<!-- It may be useful to indicate the endianness of the encoding) -->
 
 ## Reserved Namespaces
 
+Celestia reserves certain namespaces with specific meanings.
+Applications SHOULD refrain from using these reserved namespaces for their blob data. <!-- the implication of this is still under investigation, but it is prudent to advise app developers not using this namespace -->
```

Comment:
This sentence aims to provide clarity to app developers regarding the consequences of overlooking reserved namespaces and using them for their blobs.

>Isn't it straight up infeasible to use these reserved namespaces

Can you please expand on this part? Why it is infeasible? Is it part of the transaction validity rules? where this check is done? What happens if a blob's namespace has a reserved namespace?

Reply:
Ok, found the answer in the code, changed `SHOULD` to `MUST` in https://github.com/celestiaorg/celestia-app/pull/2124/commits/f171508ce374deda53303124ca7f6b3235498fed.

Reply:
One question @rootulp: The expected behavior is to reject blob transactions with reserved namespaces? right? asking this for the code audit's sake

Reply:
Yes, it would be rejected in ProcessProposal

Reply:
> The expected behavior is to reject blob transactions with reserved namespaces?

Yes.

Reply:
> Yes, it would be rejected in ProcessProposal

If it is checked during `ProcessProposal`, then how would the tx sender be informed that its transaction was invalid? or they are not informed?